src/OrchardCore/OrchardCore/Modules/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -157,20 +156,23 @@ private static void AddAntiForgery(OrchardCoreBuilder builder)
             builder.ConfigureServices((services, serviceProvider) =>
             {
                 var settings = serviceProvider.GetRequiredService<ShellSettings>();
-                var httpContextAccessor = serviceProvider.GetRequiredService<IHttpContextAccessor>();
 
                 var tenantName = settings.Name;
-                var tenantPrefix = "/" + settings.RequestUrlPrefix;
 
-                // 'PathBase' doesn't include any tenant prefix when called from 'ConfigureServices()'.
-                var pathBase = httpContextAccessor.HttpContext?.Request.PathBase ?? new PathString(String.Empty);
+                // Re-register the antiforgery  services to be tenant-aware.
+                var collection = new ServiceCollection()
+                    .AddAntiforgery(options =>
+                    {
+                        options.Cookie.Name = "orchantiforgery_" + tenantName;
 
-                services.AddAntiforgery(options =>
-                {
-                    options.Cookie.Name = "orchantiforgery_" + tenantName;
-                    // Use 'PathBase' if e.g under a virtual folder.
-                    options.Cookie.Path = pathBase + tenantPrefix;
-                });
+                        // Don't set the cookie builder 'Path' so that it uses the 'IAuthenticationFeature' value
+                        // set by the pipeline and comming from the request 'PathBase' which already ends with the
+                        // tenant prefix but may also start by a path related e.g to a virtual folder.
+
+                        // options.Cookie.Path = tenantPrefix;
```

Comment:
Same remark, I don't think it's worth having commented code that uses variables that no longer exist :smile:

Reply:
Okay